experimental/module-declarations.md

```diff
@@ -0,0 +1,68 @@
+# [JavaScript Module Declarations][proposal-module-declarations]
+
+## Module block
+
+```js
+interface ModuleBlock <: Node {
+    type: "ModuleBlock";
+    sourceType: "module";
```

Comment:
I chose `ModuleBlock` here because it starts with `{` and end with `}`, similar to `BlockStatement` and `StaticBlock`, so I slightly prefer `ModuleBlock` over `ModuleBody`.

The `sourceType` property is a redundant property so that `ModuleBlock` share the structure with `Program`. It will be convenient for AST consumers: scope queries like `scope.getProgramParent().sourceType` just work.